pkg/asset-manager-utils/test/RebalancingRelayer.test.ts

```diff
@@ -0,0 +1,284 @@
+    });
+    poolId = await pool.getPoolId();
+  });
+
+  describe('vault', () => {
+    it('uses the given vault', async () => {
+      expect(await relayer.vault()).to.be.equal(vault.address);
+    });
+  });
+
+  describe('join', () => {
+    let request: { assets: string[]; maxAmountsIn: BigNumberish[]; userData: string; fromInternalBalance: boolean };
+
+    sharedBeforeEach('build join request', async () => {
+      const amountsIn = Array(tokens.length).fill(fp(10));
+      request = {
+        assets: tokens.addresses,
+        maxAmountsIn: amountsIn,
+        userData: encodeJoinWeightedPool({ kind: 'Init', amountsIn }),
+        fromInternalBalance: false,
+      };
+    });
+
+    context('when going through the relayer', () => {
+      context('when the relayer is allowed to join', () => {
+        sharedBeforeEach('allow relayer', async () => {
+          const action = await actionId(vault, 'joinPool');
+          await authorizer.connect(admin).grantRole(action, relayer.address);
+        });
+
+        context('when the user did allow the relayer', () => {
+          sharedBeforeEach('allow relayer', async () => {
+            await vault.connect(sender).setRelayerApproval(sender.address, relayer.address, true);
+          });
+
+          it('joins the pool', async () => {
+            const previousSenderBalance = await pool.balanceOf(sender.address);
+            const previousRecipientBalance = await pool.balanceOf(recipient.address);
+            const previousRelayerBalance = await pool.balanceOf(relayer.address);
+
+            const receipt = await relayer.connect(sender).joinPool(poolId, recipient.address, request);
+
+            expectEvent.inIndirectReceipt(await receipt.wait(), pool.interface, 'Join', {
+              poolId,
+              sender: sender.address,
+              recipient: recipient.address,
+              userData: request.userData,
+            });
+
+            const currentSenderBalance = await pool.balanceOf(sender.address);
+            expect(currentSenderBalance).to.be.equal(previousSenderBalance);
+
+            const currentRecipientBalance = await pool.balanceOf(recipient.address);
+            expect(currentRecipientBalance.gt(previousRecipientBalance)).to.be.true;
+
+            const currentRelayerBalance = await pool.balanceOf(relayer.address);
+            expect(currentRelayerBalance).to.be.equal(previousRelayerBalance);
+          });
+
+          it('rebalances the pool', async () => {
+            const receipt = await relayer.connect(sender).joinPool(poolId, recipient.address, request);
+
+            expectEvent.inIndirectReceipt(await receipt.wait(), assetManagers[0].interface, 'Rebalanced', {
+              poolId,
+              assetManager: assetManagers[0].address,
+              token: tokens.first.address,
+              force: false,
+            });
+
+            expectEvent.inIndirectReceipt(await receipt.wait(), assetManagers[0].interface, 'Rebalanced', {
```

Comment:
```suggestion
            expectEvent.inIndirectReceipt(await receipt.wait(), assetManagers[1].interface, 'Rebalanced', {
```

Really the same, but slightly more correct :)